cmd/helm/repo_add.go

```diff
@@ -160,10 +160,11 @@ func (o *repoAddOptions) run(out io.Writer) error {
 	// 2. When the config is different require --force-update
 	if !o.forceUpdate && f.Has(o.name) {
 		existing := f.Get(o.name)
-		if c != *existing {
-
-			// The input coming in for the name is different from what is already
-			// configured. Return an error.
+		// In case of of the repo URLs ends with a trailing slash and the other
+		// one doesn't, the config is considered to be different even though they
+		// refer to the same repository.
+		// Only fail with an error if the configs actually are different.
+		if c != *existing && c.URLWithTrailingSlash() != existing.URLWithTrailingSlash() {
```

Comment:
If we're already comparing the same URLs with a trailing slash, why do we need the first comparator? Effectively you're comparing the same URL twice. The first check is no longer required.

Reply:
Instead of adding `c.URLWithTrailingSlash() != existing.URLWithTrailingSlash()` check, You can set URL's of both the objects  before line `162` as below

```
c.url = c.URLWithTrailingSlash()
existing.url = existing.URLWithTrailingSlash()
```

In this manner single check `c != *existing` is sufficient.

Reply:
> If we're already comparing the same URLs with a trailing slash, why do we need the first comparator?

If the URL is the only thing that musn't exist already, I'll just remove the `c != existing` condition. I didn't know the exact business rules at this point.

> You can set URL's of both the objects before line 162 as below

This implies that all repo URLs will be processed with a trailing slash even if no trailing slash has been specified by the user. Would this be ok or should we just use the `c.URLWithTrailingSlash() != existing.URLWithTrailingSlash()` comparison as it is, @bacongobbler?

Reply:
@dominikbraun Although all repo URLs will be processed with a trailing slash even if no trailing slash has been specified by the user but these changes won't be persisted. See line no. 172.

Reply:
> This implies that all repo URLs will be processed with a trailing slash even if no trailing slash has been specified by the user. Would this be ok or should we just use the `c.URLWithTrailingSlash() != existing.URLWithTrailingSlash()` comparison as it is, @bacongobbler?

Yes, this is fine. Consider the test cases we are attempting to catch and their expected outcomes:

1. repository URLs match, neither having a trailing slash - no-op
1. repository URLs match, one having a trailing slash - no-op
1. repository URLs match, both having a trailing slash - no-op
1. repository URLs do not match, neither having a trailing slash - fail
1. repository URLs do not match, one having a trailing slash - fail
1. repository URLs do not match, both having a trailing slash - fail

Consider the question: does removing the check result in the same expected behaviour? If yes, then it's fine to remove. In this case, yes it is fine to remove and only process these URLs with a trailing slash - the end result is the same.